packages/utils/src/index.js

```diff
@@ -1,7 +1,32 @@
 // @flow
 import type { RegisteredCache, EmotionCache, SerializedStyles } from './types'
 
-let isBrowser = typeof document !== 'undefined'
+const isBrowser = typeof document !== 'undefined'
+
+const getGlobalInstances = (context: any) => {
+  const globalVersionsKey = '__EMOTION_VERSIONS__'
+  if (!context[globalVersionsKey]) {
+    context[globalVersionsKey] = {}
+  }
+  return context[globalVersionsKey]
+}
+
+type ModuleRegister = { [key: string]: boolean }
+
+const globalContext = isBrowser ? window : global
+
+const instances: ModuleRegister = getGlobalInstances(globalContext)
+
+export function registerEmotionModule(name: string) {
```

Comment:
I don't think this abstraction is much needed, we only need this for a single module (`@emotion/core`), so I would prefer avoiding extra abstractions and just inlining this stuff to `@emotion/core`'s index

I also don't think we really need tests for this - it's rather hard to break it and writing a proper test (without testing implementation details) for this thing would be rather brittle and complicated